flink-external-resources/flink-external-resource-gpu/src/main/java/org/apache/flink/externalresource/gpu/GPUDriver.java

```diff
@@ -71,16 +74,16 @@
 		final String discoveryScriptPath = config.getString(DISCOVERY_SCRIPT_PATH);
 		if (StringUtils.isNullOrWhitespaceOnly(discoveryScriptPath)) {
 			throw new IllegalConfigurationException(
-				"Could not find config of the path of gpu discovery script, the relevant config key is {}.", "external-resource.<resource-name>." + DISCOVERY_SCRIPT_ARG.key());
+				String.format("GPU discovery script ('%s') is not configured.", ExternalResourceOptions.keyWithResourceNameAndSuffix("<resource-name>", DISCOVERY_SCRIPT_ARG.key())));
```

Comment:
I would suggest to make `ExternalResourceOptions#genericKeyWithSuffix` public and use it, rather than passing in `<resource-name>`.